File: src/main/java/kr/co/bookand/backend/bookstore/domain/BookStoreImage.java
package kr.co.bookand.backend.bookstore.domain;

import lombok.*;

import javax.persistence.*;

@Entity
@Getter
@Builder
@AllArgsConstructor
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class BookStoreImage {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column()
    private Long id;
    private String url;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "bookstore_id")
    private BookStore bookStore;

    public void updateBookStore(BookStore bookStore) {
        this.bookStore = bookStore;
    }
}

File: src/main/kotlin/kr/co/bookand/backend/bookstore/model/BookstoreImage.kt
package kr.co.bookand.backend.bookstore.model

import javax.persistence.*

@Entity
class BookstoreImage(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "bookstore_image_id")
    var id: Long = 0,

    var url: String,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "bookstore_id")
    var bookstore: Bookstore? = null
) {
    fun updateBookStore(bookstore: Bookstore) {
        this.bookstore = bookstore
    }
}